src/bio/bio_xstream.c

```diff
@@ -893,6 +896,11 @@ bio_xsctxt_free(struct bio_xs_context *ctxt)
 		ctxt->bxc_desc = NULL;
 	}
 
+	if (ctxt->bxc_blobstore->bb_dev_health->bhm_desc != NULL) {
+		spdk_bdev_close(ctxt->bxc_blobstore->bb_dev_health->bhm_desc);
+		ctxt->bxc_blobstore->bb_dev_health->bhm_desc = NULL;
+	}
+
```

Comment:
The open descriptor should be closed by the owner xstream who opened it, also the io channel needs be put. So you'd call bio_fini_health_monitoring() for owner xstream only.

Reply:
bio_fini_health_monitoring() would be called when bio blobstore is being freed, correct? now the bdev desc would be closed in bio_fini_health_monitoring() as well, so nothing would be called from bio_xsctxt_free()